nexus/types/src/deployment/blueprint_diff.rs

```diff
@@ -85,6 +692,58 @@ impl ZoneSortKey for ModifiedZone {
 }
 
 impl ModifiedZone {
+    pub fn from_diff(
+        diff: &BlueprintZoneConfigDiff,
+    ) -> Result<ModifiedZone, BpDiffZoneError> {
+        // Do we have any errors? If so, create a "reason" string.
+        let mut reason = String::new();
+        // These first two checks are only for backwards compatibility. They are
+        // all included in the zone_type comparison below.
```

Comment:
Can we drop these two since they're covered by the later `zone_type` check? If that changes the diffs from tests, maybe do that as a small followup PR?

Reply:
I anticipate a few cleanup tasks in a follow up that break backwards compatibility. I'll add this one.